examples/api/lib/material/app/app.0.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+/// Flutter code sample for [MaterialApp].
+
+void main() {
+  runApp(const MaterialAppExample());
+}
+
+class MaterialAppExample extends StatefulWidget {
+  const MaterialAppExample({super.key});
+
+  @override
+  State<MaterialAppExample> createState() => _MaterialAppExampleState();
+}
+
+class _MaterialAppExampleState extends State<MaterialAppExample> {
+  bool isDarkTheme = false;
+  bool isCustomizedThemeAnimation = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      themeMode: isDarkTheme ? ThemeMode.dark : ThemeMode.light,
+      theme: ThemeData(colorScheme: ColorScheme.fromSeed(
+        seedColor: Colors.green,
+      )),
+      darkTheme: ThemeData(colorScheme: ColorScheme.fromSeed(
+        seedColor: Colors.green,
+        brightness: Brightness.dark,
+      )),
+      themeAnimationStyle: !isCustomizedThemeAnimation ? null : AnimationStyle(
```

Comment:
A segmented button labeled AnimationStyle: with three options: `default, custom, none` would also show devs how to turn the animation off (a feature that's been asked for)